tests/unit/test-util/test.artifacts.js

```diff
@@ -55,4 +55,74 @@ describe('prepareArtifactsDir', () => {
     }
   ));
 
+  it('throws an UsageError when it lacks permissions to stat the directory',
+    () => withTempDir(
+      function(tmpDir) {
+        if (process.platform === 'win32') {
+          // Skipped on windows because the mkdir permissions do not have any effect (web-ext#776)
+          this.skip();
+          return;
+        }
+        const tmpPath = path.join(tmpDir.path(), 'build');
+        return fs.mkdir(tmpPath, '0622').then(() => {
+          const artifactsDir = path.join(tmpPath, 'artifacts');
+          return prepareArtifactsDir(artifactsDir)
+            .then(makeSureItFails())
+            .catch(onlyInstancesOf(UsageError, (error) => {
+              assert.match(error.message, /Cannot access.*lacks permissions/);
+            }));
+        });
+      }
+  ));
+
+  it('throws error when directory exists but lacks writing permissions',
+    () => withTempDir(
+      function(tmpDir) {
+        if (process.platform === 'win32') {
+          // Skipped on windows because the mkdir permissions do not have any effect (web-ext#776)
+          this.skip();
+          return;
+        }
+        const artifactsDir = path.join(tmpDir.path(), 'dir-nowrite');
+        return fs.mkdir(artifactsDir, '0555').then(() => {
+          return prepareArtifactsDir(artifactsDir)
+            .then(makeSureItFails())
+            .catch(onlyInstancesOf(UsageError, (error) => {
+              assert.match(error.message, /exists.*lacks permissions/);
+            }));
+        });
+      }
+  ));
+
+  it('throws error when creating a folder if lacks writing permissions',
+    () => withTempDir(
```

Comment:
This has the same problem as above. It needs to begin with `function` so that `this` gets bound to the test instance.